pkg/api/middlewares.go

```diff
@@ -27,3 +29,37 @@ func getHTTPCacheGenNumberHeaderSetterMiddleware(cacheGenNumbersLoader *purger.T
 		})
 	})
 }
+
+// HTTPHeaderMiddleware adds specified HTTPHeaders to the request context
+type HTTPHeaderMiddleware struct {
+	TargetHeaders []string
+}
+
+// InjectTargetHeadersIntoHTTPRequest injects specified HTTPHeaders into the request context
+func (h HTTPHeaderMiddleware) InjectTargetHeadersIntoHTTPRequest(r *http.Request) context.Context {
+	headerMap := make(map[string]string)
+
+	// Check to make sure that Headers have not already been injected
+	testing, ok := r.Context().Value(util_log.HeaderMapContextKey).(map[string]string)
+	if ok && h.TargetHeaders != nil && testing[h.TargetHeaders[0]] == r.Header.Get(h.TargetHeaders[0]) {
```

Comment:
Hm i think testing if the context already exist in the context is enough. I would drop the check from headers.